loopout.go

```diff
@@ -456,9 +456,11 @@ func (s *loopOutSwap) handlePaymentResult(result paymentResult) error {
 		return nil
 
 	case result.status.State == lnrpc.Payment_SUCCEEDED:
-		s.cost.Server += result.status.Value.ToSatoshis() -
-			s.AmountRequested
-		s.cost.Offchain += result.status.Fee.ToSatoshis()
+		if result.status.Preimage == s.SwapContract.Preimage {
```

Comment:
Please add a comment about this check (about the scenarios).

Reply:
Instead I added a flag to indicate this is a swap payment and some comments explaining what we're doing when calculating the costs.